editor/translations/translation_catalog_maker.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import shutil
+from collections import OrderedDict
+
+EXTRACT_TAGS = ["description", "brief_description", "member", "constant", "theme_item", "link"]
+HEADER = '''\
+# LANGUAGE translation of the Godot Engine editor
+# Copyright (c) 2007-2020 Juan Linietsky, Ariel Manzur.
+# Copyright (c) 2014-2020 Godot Engine contributors (cf. AUTHORS.md).
+# This file is distributed under the same license as the Godot source code.
+#
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: Godot Engine editor\\n"
+"Content-Type: text/plain; charset=UTF-8\\n"
+"Content-Transfer-Encoding: 8-bit\\n"
+
+'''
+
+## <xml-line-number-hack from="https://stackoverflow.com/a/36430270/10846399">
+import sys
+sys.modules['_elementtree'] = None
+import xml.etree.ElementTree as ET
+
+## override the parser to get the line number 
+class LineNumberingParser(ET.XMLParser):
+    def _start(self, *args, **kwargs):
+        ## Here we assume the default XML parser which is expat
+        ## and copy its element position attributes into output Elements
+        element = super(self.__class__, self)._start(*args, **kwargs)
+        element._start_line_number = self.parser.CurrentLineNumber
+        element._start_column_number = self.parser.CurrentColumnNumber
+        element._start_byte_index = self.parser.CurrentByteIndex
+        return element
+
+    def _end(self, *args, **kwargs):
+        element = super(self.__class__, self)._end(*args, **kwargs)
+        element._end_line_number = self.parser.CurrentLineNumber
+        element._end_column_number = self.parser.CurrentColumnNumber
+        element._end_byte_index = self.parser.CurrentByteIndex
+        return element
+## </xml-line-number-hack>
+
+class Desc:
+	def __init__(self, line_no, msg, desc_list=None):
+		## line_no : the line number where the desc is
+		self.line_no = line_no
+		self.msg = msg
+		self.desc_list = desc_list
+
+class DescList:
+	def __init__(self, doc, path):
+		## doc  : root xml element of the document
+		## path : file path of the xml document
+		## list : list of Desc objects for this document
+		self.doc = doc
+		self.path = path
+		self.list = []
+
+def print_error(error):
+	print("ERROR: {}".format(error))
+    
+## build classes with xml elements recursively
+def _collect_classes_dir(path, classes):
+	if not os.path.isdir(path):
```

Comment:
Mixed indentation styles (2 spaces, 4 spaces). It should be 4 spaces everywhere.